### DISCUSSION TITLE
How to you join public channels and listen to messages on them (as a user)?

### DISCUSSION THREAD

**Discussion by lessless** [visciang/telegram#166] [Q&A] [answered]
Hi,

I want to validate an app idea that requires listening to messages across a list of channels.
According to what I was able to find, it's only possible with "user bots" https://stackoverflow.com/questions/42674340/how-to-join-my-telegram-bot-to-public-channel

Here is what JunctionBot says in it's readme for example: 

> If you are admin of the source channel, keep in mind that the bot will enter there with one of its accounts. Do not ban the bot, otherwise it will not be able to get messages from there.
> 
> Now send me a link to a destination channel, to where you want to copy messages.
> - For public channels, you can send me a channel @name. It starts with the symbol `@`. Please note that you cannot use the full descriptive name of the channel, you need exactly that short @name, by clicking on which you can go to the channel itself.
> - For private channels, send me an invitation link.
> - If you know the channel ID, you can send it to me as a number.
> 

Is that currently possible with the telegram library? 

Thanks!

**Comment by visciang** [comment, marked as answer]
I've no experience with it, sorry.
Not sure if the telegram HTTP Bot API support something for this

**Reply by lessless** [reply]
Yeah! It seems that require MTProto-based integration https://github.com/LonamiWebs/Telethon/wiki/MTProto-vs-HTTP-Bot-API
